File: src/main/java/net/matees/Arcade.java
package net.matees;

import java.util.List;

import org.bukkit.event.Listener;
import org.bukkit.plugin.java.JavaPlugin;

import me.kodysimpson.simpapi.command.CommandManager;
import me.kodysimpson.simpapi.menu.MenuManager;
import net.matees.arcade.Minigame;
import net.matees.arcade.itemrush.ItemRush;
import net.matees.arcade.lavarise.LavaRise;
import net.matees.arcade.manhunt.Manhunt;
import net.matees.arcade.mobrush.MobRush;
import net.matees.commands.OpenArcadeMenu;
import net.matees.commands.OpenGlobalSettingsMenu;
import net.matees.settings.global.GlobalSettings;

public final class Arcade extends JavaPlugin {
    private static Arcade plugin;
    private Minigame currentGame;

    private List<Minigame> minigames;

    @Override
    public void onEnable() {
        // Plugin startup logic
        plugin = this;
        MenuManager.setup(getServer(), this);

        try {
            initalizeCommands();
        } catch (NoSuchFieldException | IllegalAccessException e) {
            e.printStackTrace();
        }
        setupArcade();
    }

    private void initalizeCommands() throws NoSuchFieldException, IllegalAccessException {
        CommandManager.createCoreCommand(this, "arcade", "Arcade plugin - Survival minigames", "/arcade open|settings",
                null, OpenArcadeMenu.class, OpenGlobalSettingsMenu.class);
    }

    private void setupArcade() {
        minigames = List.of(
                ItemRush.getInstance(),
                MobRush.getInstance(),
                Manhunt.getInstance(),
                LavaRise.getInstance());

        for (Listener listener : GlobalSettings.getInstance().getListeners()) {
            getServer().getPluginManager().registerEvents(listener, this);
        }
    }

    @Override
    public void onDisable() {
        // Plugin shutdown logic
    }

    public static Arcade getPlugin() {
        return plugin;
    }

    public Minigame getCurrentGame() {
        return this.currentGame;
    }

    public void setCurrentGame(Minigame currentGame) {
        this.currentGame = currentGame;
    }

    public List<Minigame> getMinigames() {
        return minigames;
    }
}

File: src/main/java/net/matees/Arcade.kt
package net.matees

import me.kodysimpson.simpapi.command.CommandManager
import me.kodysimpson.simpapi.menu.MenuManager
import net.matees.arcade.Minigame
import net.matees.arcade.deathswap.DeathSwap
import net.matees.arcade.blockshuffle.BlockShuffle
import net.matees.arcade.itemrush.ItemRush
import net.matees.arcade.lavarise.LavaRise
import net.matees.arcade.mobrush.MobRush
import net.matees.commands.OpenArcadeMenu
import net.matees.commands.OpenGlobalSettingsMenu
import net.matees.settings.global.GlobalSettings
import org.bukkit.plugin.java.JavaPlugin

class Arcade : JavaPlugin() {
    var currentGame: Minigame? = null

    var minigames: List<Minigame>? = null
        private set

    override fun onEnable() {
        // Plugin startup logic
        plugin = this
        MenuManager.setup(server, this)

        try {
            initializeCommands()
        } catch (e: NoSuchFieldException) {
            e.printStackTrace()
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }
        setupArcade()

        logger.info("Started Arcade")
    }

    @Throws(NoSuchFieldException::class, IllegalAccessException::class)
    private fun initializeCommands() {
        CommandManager.createCoreCommand(
            this, "arcade", "Arcade plugin - Survival minigames", "/arcade open|settings",
            null, OpenArcadeMenu::class.java, OpenGlobalSettingsMenu::class.java
        )
    }

    private fun setupArcade() {
        minigames = listOf<Minigame>(
            ItemRush.instance,
            MobRush.instance,  // Manhunt.getInstance(), WIP
            LavaRise.instance,
            DeathSwap.instance,
            BlockShuffle.instance
        )

        for (listener in GlobalSettings.Companion.instance.listeners) {
            server.pluginManager.registerEvents(listener, this)
        }
    }

    override fun onDisable() {
        // Plugin shutdown logic
    }

    companion object {
        var plugin: Arcade? = null
            private set
    }
}
